src/main/java/de/hysky/skyblocker/skyblock/dungeon/CroesusProfit.java

```diff
@@ -0,0 +1,267 @@
+package de.hysky.skyblocker.skyblock.dungeon;
+
+import com.google.gson.JsonObject;
+import de.hysky.skyblocker.config.SkyblockerConfigManager;
+import de.hysky.skyblocker.skyblock.item.tooltip.TooltipInfoType;
+import de.hysky.skyblocker.utils.ItemUtils;
+import de.hysky.skyblocker.utils.render.gui.ColorHighlight;
+import de.hysky.skyblocker.utils.render.gui.ContainerSolver;
+import net.minecraft.item.ItemStack;
+import net.minecraft.text.Text;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class CroesusProfit extends ContainerSolver {
+    private static final Pattern ESSENCE_PATTERN = Pattern.compile("(?<type>[A-Za-z]+) Essence x(?<amount>[0-9]+)");
+    public CroesusProfit() {
+        super(".*Catacombs - Floor.*");
+    }
+
+    @Override
+    protected boolean isEnabled() {
+        return SkyblockerConfigManager.get().locations.dungeons.dungeonChestProfit.croesusProfit;
+    }
+
+    @Override
+    protected List<ColorHighlight> getColors(String[] groups, Map<Integer, ItemStack> slots) {
+        List<ColorHighlight> highlights = new ArrayList<>();
+        ItemStack bestChest = null, secondBestChest = null;
+        long bestValue = 0, secondBestValue = 0;    // If negative value of chest - it is out of the question
+        long dungeonKeyPriceData = getItemPrice("DUNGEON_CHEST_KEY") * 2; // lesser ones don't worth the hassle
```

Comment:
Why are you multiplying by two? Just a bit confused.

Reply:
just to resist possible market fluctuations that sometimes happen - I witnessed that crap twice while I were trying out code

Reply:
also it is here to prevent cases when chest is like 1 coin profit after deducting key price

Reply:
Oh okay.